lib/fetchHttpClient.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import "isomorphic-fetch";
+import * as tough from "tough-cookie";
+import AbortController from "abort-controller";
+import FormData from "form-data";
+
+import { HttpClient } from "./httpClient";
+import { WebResource } from "./webResource";
+import { HttpOperationResponse } from "./httpOperationResponse";
+import { HttpHeaders } from "./httpHeaders";
+import { RestError } from "./restError";
+import { Readable, Transform } from "stream";
+
+interface FetchError extends Error {
+    code: string | undefined;
+    errno: string | undefined;
+    type: string | undefined;
+}
+
+export class FetchHttpClient implements HttpClient {
+    private readonly cookieJar = new tough.CookieJar();
+
+    async sendRequest(httpRequest: WebResource): Promise<HttpOperationResponse> {
+        if (!httpRequest) {
```

Comment:
We aren't checking the type of httpRequest. Someone can pass a boolean and the condition will still pass in JS.

`if (typeof httpRequest !== "object") {` // This will check for null, undefined cases and also does the type checking.

Reply:
You probably already looked at this but:

```js
// in browser REPL
> typeof null
< "object"
```